### PR TITLE
[CIP 515] sifive blocks: Counter --> Timer

### PR DESCRIPTION
**Pull request by mhtwn**

### CODE REVIEW

**Comment by mhtwn** [comment]
I have no idea how the other commit (0cabd4e) is appearing in this PR - when I looked at master it seems that it never went through.

File: src/main/scala/util/Timer.scala [hunk 1]
```diff
@@ -43,7 +43,7 @@ case class GenericTimerCfgDescs(
 
 object DefaultGenericTimerCfgDescs  {
   def apply(prefix: String, ncmp: Int): GenericTimerCfgDescs = GenericTimerCfgDescs(
-    scale = RegFieldDesc(s"${prefix}scale", "Counter scale value."),
+    scale = RegFieldDesc(s"${prefix}scale", "Timer scale value."),
```
Comment (mwachs5) [inline src/main/scala/util/Timer.scala]: this should still say Counter

File: src/main/scala/util/Timer.scala [hunk 2]
```diff
@@ -185,12 +185,12 @@ trait GenericTimer {
   require(ncmp > 0)
 
   protected def countLo_desc: RegFieldDesc = if (countWidth > regWidth) {
-    RegFieldDesc(s"${prefix}countlo", "Low bits of Counter", volatile=true)
+    RegFieldDesc(s"${prefix}countlo", "Low bits of Timer", volatile=true)
```
Comment (mwachs5) [inline src/main/scala/util/Timer.scala]: This should still say Counter

File: src/main/scala/util/Timer.scala [hunk 3]
```diff
@@ -185,12 +185,12 @@ trait GenericTimer {
...
   } else  {
-    RegFieldDesc(s"${prefix}count", "Counter Register", volatile=true)
+    RegFieldDesc(s"${prefix}count", "Timer Register", volatile=true)
```
Comment (mwachs5) [inline src/main/scala/util/Timer.scala]: this should still say Counter

File: src/main/scala/util/Timer.scala [hunk 4]
```diff
@@ -185,12 +185,12 @@ trait GenericTimer {
...
   }
-  protected def countHi_desc: RegFieldDesc = if (countWidth > regWidth) RegFieldDesc(s"${prefix}counthi", "High bits of Counter", volatile=true) else RegFieldDesc.reserved
-  protected def s_desc: RegFieldDesc = RegFieldDesc(s"${prefix}s", "Scaled value of Counter", access=RegFieldAccessType.R, volatile=true)
+  protected def countHi_desc: RegFieldDesc = if (countWidth > regWidth) RegFieldDesc(s"${prefix}counthi", "High bits of Timer", volatile=true) else RegFieldDesc.reserved
+  protected def s_desc: RegFieldDesc = RegFieldDesc(s"${prefix}s", "Scaled value of Timer", access=RegFieldAccessType.R, volatile=true)
```
Comment (mwachs5) [inline src/main/scala/util/Timer.scala]: This should still say Counter

**Comment by mhtwn** [comment]
> There seems to be a lot more commits in here than necessary...? though i like the last one!

Some Edits are for clocks in DTS (and those are from my previous PR) - I am not sure why those were unmerged till today.

**Comment by johningalls-sifive** [comment]
please update Federation master to point to sifive-blocks/master before deleting the branch

**Pull request merged**